R/module_teal_navigation.R

```diff
@@ -491,3 +462,123 @@ srv_teal_module.teal_module <- function(id,
+    src = "module-navigation",
+    stylesheet = "module-navigation.css",
+    script = "module-navigation.js"
+  )
+}
+
+#' Create Bootstrap based Navigation for Teal Modules
+#'
+#' Generates a dropdown navigation interface that allows users to switch
+#' between different teal modules. The function creates both the navigation dropdown menu
+#' and the corresponding tab content containers.
+#'
+#' @details
+#' This function constructs a complete navigation system with the following components:
+#' - A dropdown button labeled "Modules" that reveals module options on hover
+#' - Individual navigation links for each module, grouped by their `group` attribute
+#' - Tab content containers that house the actual module UI elements
+#' - Automatic tab switching via Bootstrap's tab functionality
+#'
+#' The navigation leverages `shiny`'s built-in tab system by using the `shiny-tab-input` class,
+#' which automatically handles showing/hiding content when navigation links are clicked.
+#' Each module's UI is wrapped in a `.tab-pane` container with an ID that corresponds to
+#' the navigation link's `href` attribute.
+#'
+#' Module grouping is supported - when modules belong to different groups, visual separators
+#' and group labels are automatically inserted in the dropdown menu.
+#'
+#' @param id (`character(1)`) Unique identifier for the navigation widget. This ID is used
+#'   to create the Shiny input binding that tracks which module is currently active.
+#' @param modules (`teal_modules`) A teal modules object containing the module definitions.
+#'   Each module should have `label` and optionally `group` attributes.
+#' @param modules_ui (`named list`) List of UI elements corresponding to each module.
+#'   Names must match the module IDs from the `modules` parameter.
+#' @keywords internal
+.teal_custom_nav <- function(id, modules, modules_ui) {
+  active_module_id <- shiny::restoreInput(id, default = names(modules)[1])
+  last_group <- modules[[1]]$group
+  tags$div(
+    class = "teal-modules-wrapper",
+    .teal_custom_nav_deps(),
+    tags$ul(
+      id = id,
+      style = "align-items: center;",
+      class = "nav shiny-tab-input",
+      `data-tabsetid` = "test",
+      tags$div(
+        class = "dropdown nav-item-custom",
+        onmouseover = "initNavigationMouseOver.call(this)",
+        onmouseout = "initNavigationMouseOut.call(this)",
+        tags$a(
+          class = "dropdown-toggle active",
+          role = "button",
+          style = "text-decoration: none; border-bottom-color: #0d6efd;",
+          "Modules"
+        ),
+        tags$div(
+          class = "dropdown-menu",
+          !!!lapply(names(modules), function(module_id) {
+            module <- modules[[module_id]]
+            ui <- tags$span(
+              if (!identical(module$group, last_group) && !is.null(module$group)) {
+                tags$span(
+                  tags$br(), tags$br(),
+                  tags$span(paste(module$group, collapse = " > ")),
```

Comment:
this span could have a class to change its look. I think it could be more visible

<img width="689" alt="image" src="https://github.com/user-attachments/assets/a968e3d0-d7a5-4d3b-b08b-d39b23be1065" />

Reply:
Created an [issue](https://github.com/insightsengineering/teal/issues/1551) to track this in the future.